internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -467,6 +479,114 @@ func (r *BMACReconciler) findBMH(ctx context.Context, agent *adiiov1alpha1.Agent
 	return nil, nil
 }
 
+func (r *BMACReconciler) isDay2Worker(agent *adiiov1alpha1.Agent) bool {
+	return agent.Spec.Role == models.HostRoleWorker && agent.Spec.ClusterDeploymentName != nil
```

Comment:
What distinguishes a day1 agent that's already part of a cluster from a day2 agent? Is there something more I can check here?

Reply:
I think a better way will be to check if the ClusterDeployment it belongs to is Installed.
Yet, we need a way to know if this agent is not part of the original Day1 cluster as for now they are not deleted.

@filanov FYI

Reply:
Do we really care? if you are making sure the BMH exist then it will exist for day1 agents anyway no?